cpp/src/arrow/memory_pool.h

```diff
@@ -175,6 +176,25 @@ ARROW_EXPORT Status jemalloc_memory_pool(MemoryPool** out);
 ARROW_EXPORT
 Status jemalloc_set_decay_ms(int ms);
 
+/// \brief Get basic allocation statistics from jemalloc
+/// See the MALLCTL NAMESPACE section in jemalloc project documentation for
+/// available stats.
+Status jemalloc_get_stat(const char* name, size_t& out);
+Status jemalloc_get_stat(const char* name, uint64_t& out);
+
+/// \brief Get basic allocation statistics from jemalloc by passing a pointer
+/// This is useful for avoiding the overhead of repeated copy calls.
+Status jemalloc_get_statp(const char* name, uint64_t& out);
+
+/// \brief Resets the counter for bytes allocated in the calling thread to zero.
+/// This affects subsequent calls to thread.peak.read, but not the values returned by
+/// thread.allocated or thread.deallocated.
+Status jemalloc_peak_reset();
+
+/// \brief Prints summary statistics in human-readable form. See malloc_stats_print
```

Comment:
Let's keep the brief form brief. Also, where does it print to? Can it be changed?
```suggestion
/// \brief Print summary statistics in human-readable form.
///
/// See malloc_stats_print
```

Reply:
This prints to stdout and while `malloc_stats_print` gives other options we probably don't need them at the moment.

Reply:
Apparently it would print to stderr rather than stdout?

Reply:
Ah yes, it's stderr indeed!